### PR TITLE
AmdPlatformPkg/DynamicTables: Adds CPU and PCI SSDT tables

### PR DESCRIPTION
**Pull request by abdattar**

### CODE REVIEW

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c [hunk 1]
```diff
@@ -0,0 +1,1540 @@
...
+    Status = gBS->OpenProtocol (
+                    HandleBuffer[Index],
+                    &gEfiPciRootBridgeIoProtocolGuid,
+                    (VOID **)&Io,
+                    mDriverHandle,
+                    NULL,
+                    EFI_OPEN_PROTOCOL_GET_PROTOCOL
+                    );
+    if (EFI_ERROR (Status)) {
+      return Status;
```
Comment (changab) [inline Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:266]: You may want to free HandleBuffer in all of the error conditions in this function.

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c [hunk 2]
```diff
@@ -0,0 +1,1540 @@
...
+    }
+
+    Status = AmlCodeGenNameInteger (
+               "_ADR",
+               (RootBridge->RootPort[RPIndex]->Device << 16) + RootBridge->RootPort[RPIndex]->Function,
+               DeviceNode,
+               NULL
+               );
+    if (EFI_ERROR (Status)) {
+      ASSERT_EFI_ERROR (Status);
```
Comment (changab) [inline Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:559]: Do we have to delete the newly created node (DeviceNode) in the fail conditions in this file?

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c [hunk 3]
```diff
@@ -0,0 +1,1540 @@
...
+      return Status;
+    }
+
+    Status = AmlCodeGenNamePackage ("_CID", DeviceNode, &PackageNode);
+    ASSERT_EFI_ERROR (Status);
+
+    // Name (_CID, EISAID("PNP0A03"))
+    Status = AmlGetEisaIdFromString ("PNP0A03", &EisaId);
+    if (EFI_ERROR (Status)) {
+      ASSERT_EFI_ERROR (Status);
```
Comment (changab) [inline Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:806]: Same here, do we have to delete PackageNode in the fail conditions?

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c [hunk 4]
```diff
@@ -0,0 +1,1540 @@
...
+    }
+
+    // Accept table if OemTableId is zero.
+    if ((OemTableId == 0) ||
+        (CompareMem (&LocalTable->OemTableId, &OemTableId, 8) == 0))
+    {
+      *Table = LocalTable;
+      return EFI_SUCCESS;
+    }
+  }
```
Comment (changab) [inline Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:1046]: we dont need to return Status here?

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c [hunk 5]
```diff
@@ -0,0 +1,1540 @@
...
+  Determine all the PCI Root Bridges and PCI root ports and install resources
+  including needed _HID, _CID, _UID, _ADR, _CRS and _PRT Nodes.
+
+  @param[in]      ImageHandle   - Standard UEFI entry point Image Handle
+  @param[in]      SystemTable   - Standard UEFI entry point System Table
+
+  @retval         EFI_SUCCESS, various EFI FAILUREs.
+**/
+EFI_STATUS
+GenerateAcpiSsdtPciTable (
```
Comment (changab) [inline Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:1061]: Do we have to clean up the memory or device node in the file conditions?

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c [hunk 6]
```diff
@@ -0,0 +1,1540 @@
...
+    if (!(LocalTable->Signature == Signature)) {
+      continue;
+    }
+
+    // Accept table if OemTableId is zero.
+    if ((OemTableId == 0) ||
+        (CompareMem (&LocalTable->OemTableId, &OemTableId, 8) == 0))
+    {
+      *Table = LocalTable;
+      return EFI_SUCCESS;
```
Comment (pbgrimes) [inline Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:1044]: return should be outside of for loop?

Comment (abdattar) [reply Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:1044]: if we found the table, then no need to continue the loop.
Hence returns immediate.

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c (hunk 1 above)
Comment (abdattar) [reply Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:266]: done.

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c (hunk 2 above)
Comment (abdattar) [reply Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:559]: done, added the exit_handler

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c (hunk 3 above)
Comment (abdattar) [reply Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:806]: done, added the exit_handler

File: Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c (hunk 5 above)
Comment (abdattar) [reply Platform/AMD/AmdPlatformPkg/DynamicTables/Library/Acpi/AcpiSsdtPciLib/AcpiSsdtPciLib.c:1061]: done, added the exit_handler

**Review by abdattar — COMMENTED** [review]
Addressed the review comments.

**Comment by abdattar** [comment]
Addressed the review comments.

**Comment by abdattar** [comment]
Please review again, updated the patch.

**Review by changab — APPROVED** [review]

**Comment by abdattar** [comment]
All review comments are addressed.
PR is approved.

**Pull request merged**